android/engine/src/main/java/org/smartregister/fhircore/engine/task/FhirCarePlanGenerator.kt

```diff
@@ -141,10 +148,9 @@ constructor(val fhirEngine: FhirEngine, val transformSupportServices: TransformS
             val outcome = value.outcomeReference.find { x -> x.reference.contains(id) }
             if (outcome != null) {
               carePlanToUpdate = carePlan.copy()
-              carePlanToUpdate?.activity?.set(
-                index,
-                value.apply { detail.status = CarePlan.CarePlanActivityStatus.COMPLETED }
-              )
+              value.detail.status = CarePlan.CarePlanActivityStatus.COMPLETED
+              value.outcomeReference.first().reference = "Task/${task.id}"
```

Comment:
Instead of changing first `outcomeReference`, may be it should change first matching the previous id